Mathlib/LinearAlgebra/Matrix/Spectrum.lean

```diff
@@ -133,6 +135,27 @@ theorem det_eq_prod_eigenvalues : det A = ∏ i, (hA.eigenvalues i : 𝕜) := by
   simp_rw [Function.comp_apply]
 #align matrix.is_hermitian.det_eq_prod_eigenvalues Matrix.IsHermitian.det_eq_prod_eigenvalues
 
+/-- *spectral theorem (alternate)* (Alternate from for convenience) A hermitian matrix can be can be
+replaced by a diagonal matrix sandwiched between the eigenvector matrices. This alternate form
+allows direct rewriting of A since: $ A = V D V⁻¹$ -/
+lemma spectral_theorem' :
+    A = hA.eigenvectorMatrix ⬝ diagonal ((↑) ∘ hA.eigenvalues) ⬝ hA.eigenvectorMatrixInv := by
```

Comment:
This isn't something to deal with in this PR, but really `Matrix.IsHermitian.eigenvectorMatrix` should be a term of type `unitary (Matrix n n 𝕜)`. It would make the proof of `Matrix.IsHermitian.rank_eq_rank_diagonal` simpler too.